tests/director/tests/application_templates_subscription_test.go

```diff
@@ -88,51 +88,37 @@ func TestSubscriptionApplicationTemplateFlow(stdT *testing.T) {
 			require.Equal(t, appTmpl.ID, *actualAppPage.Data[0].ApplicationTemplateID)
 		})
 
-		t.Run("Subscribe tenant to Application flow and add bundle", func(t *testing.T) {
+		t.Run("Application is deleted successfully in consumer subaccount as a result of unsubscription", func(t *testing.T) {
 			//GIVEN
 			subscriptionToken := token.GetClientCredentialsToken(t, ctx, conf.SubscriptionConfig.TokenURL+conf.TokenPath, conf.SubscriptionConfig.ClientID, conf.SubscriptionConfig.ClientSecret, "tenantFetcherClaims")
-
-			// Subscribe
 			createSubscription(t, ctx, httpClient, appTmpl, apiPath, subscriptionToken, subscriptionConsumerTenantID, subscriptionConsumerSubaccountID, subscriptionProviderSubaccountID)
-			defer subscription.BuildAndExecuteUnsubscribeRequest(t, appTmpl.ID, appTmpl.Name, httpClient, conf.SubscriptionConfig.URL, apiPath, subscriptionToken, conf.SubscriptionConfig.PropagatedProviderSubaccountHeader, subscriptionConsumerSubaccountID, subscriptionConsumerTenantID, subscriptionProviderSubaccountID)
```

Comment:
Most probably you've removed this defer because you explicitly execute unsubscribe?
But in case something fails after the subscription and before the explicit unsubscribe request(get apps) we'll have leftover resources as part of the subscription.
So maybe it's safer to have a defer for such cases and it should not fails because currently `subscription.BuildAndExecuteUnsubscribeRequest` is implemented not to fail if the subscription is not found for such cases and cleanup

Reply:
Mistake 😄